Eastern-Research-Group/expert-query
# Bugfix/256 latest not returning true latest

## Related Issues:
* [EQ-267](https://jira.epa.gov/browse/EQ-267)

## Main Changes:
* Fixed a bug where filters could cause the incorrect reporting cycle to be chosen for the latest query.
  * This was mainly done by removing filters from the sub query, but also by updating the reporting cycle materialized views to also have organization id, region, and state. 
  * As an optimization, I did allow the sub query to be filtered by the `region`, `state`, and `organizationId` filters.
* Added code to prevent the reporting cycle filter from using the `latest` sub query if the user selected something other than `latest` for the reporting cycle. 
  * Before we were doing something like the following if a user selected 2020 for the reporting cycle: `SELECT * FROM <profile> WHERE (organizationid, reportingcycle) IN (SELECT organizationid, max(reportingcycle) FROM <profile> WHERE reportingcycle='2020' GROUP BY organizationid`. 
  * Now we do the following: `SELECT * FROM <profile> WHERE reportingcycle = '2020'`
* Updated the Reporting Cycle field to be filtered by region, state, and organization id. 

## Steps To Test:
1. Verify envs are pointing to local database
2. Run the etl
3. Start the app 
4. Verify the latest reporting cycle filter works as expected
    * A helpful way to check this is to add `console.log('query: ', query.toString());` after line 341 in the `app/server/app/routes/attains.js` file
6. Verify selecting a reporting cycle year as the filter works as expected
7. Verify that selecting a region, state, and or organization id filters the reporting cycle options
